### PR TITLE
feat: implementing data intake component

### PR DESCRIPTION
**Pull request by drduhe**
**Issue #, if available:** n/a

### Notes
* Implementing infrastructure required for the ``osml-data-intake`` package.
* Updating various dependencies to get rid of NPM warnings.
* Updating ``Repository`` and ``Bucket`` constructs to migrate off of depreciate ``autoDelete`` flag
* Implementing unit tests for new infra
* Making a couple other bug fixes / typos elsewhere

### Checklist

Before you submit a pull request, please make sure you have the following:
- [x] Code changes are compact and well-structured to facilitate easy review
- [x] Changes are documented in the README.md and other relevant documentation pages
- [x] PR title and description accurately reflect the changes and are detailed enough for historical tracking
- [x] PR contains tests that cover all new code and the code has been manual tested
- [x] All new dependencies are declared (if any), and no unnecessary libraries are added
- [x] Performance impacts (if any) of the changes are evaluated and documented
- [x] Security implications of the changes (if any) are reviewed and addressed
- [x] I have read the [Contributing Guidelines](../CONTRIBUTING.md) and agree to follow the [Code of Conduct](../CODE_OF_CONDUCT.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/osml/data_intake/di_dataplane.ts [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+   * @param {number} LAMBDA_STORAGE_SIZE - The size of the storage to assign lambda runtime in GB.
+   * @param {string} SNS_INPUT_TOPIC_NAME - The name to give the input SNS topic.
+   * @param {string} SNS_OUTPUT_TOPIC_NAME - The name to give the output SNS topic.
+   * @param {string} S3_INPUT_BUCKET_NAME - The name to give the input bucket.
+   * @param {string} S3_OUTPUT_BUCKET_NAME - The name to give the output bucket.
+   */
+  constructor(
+    public LAMBDA_ROLE_NAME: string = "DILambdaRole",
+    public LAMBDA_FUNCTION_NAME: string = "DILambda",
+    public LAMBDA_MEMORY_SIZE: number = 1024,
```
Comment (edparris) [inline lib/osml/data_intake/di_dataplane.ts:37]: May want to increase these memory and storage size limits. Not hard to find an image that is larger than 1GB.

File: lib/osml/data_intake/di_dataplane.ts [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+      topicName: this.config.SNS_OUTPUT_TOPIC_NAME
+    });
+
+    // Define a Lambda function with a container image
+    this.lambdaFunction = new DockerImageFunction(
+      this,
+      "ImageHandlerFunction",
+      {
+        code: props.dockerImageCode,
+        timeout: Duration.seconds(120),
```
Comment (edparris) [inline lib/osml/data_intake/di_dataplane.ts:155]: Consider making this timeout a parameter. Also test it with some of the Maxar NITFs. The time to create overviews and generate image statistics for a big image might push up against this timeout.

File: lib/osml/data_intake/roles/di_lambda_role.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    )!;
+
+    // Create an AWS IAM role for the Tile Server Lambda Sweeper Function
+    const role = new Role(this, "DILambdaRole", {
+      roleName: props.roleName,
+      assumedBy: new CompositePrincipal(
+        new ServicePrincipal("lambda.amazonaws.com")
+      ),
+      description:
+        "Allows the OversightML Image Handler Lambda to access necessary AWS services (SNS, S3, ...)"
```
Comment (edparris) [inline lib/osml/data_intake/roles/di_lambda_role.ts:83]: NIT: Data Intake Handler...

File: lib/osml/data_intake/di_container.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+  /**
+   * Optional configuration for the DIContainer Construct
+   */
+  config?: DIContainerConfig;
+}
+
+/**
+ * Represents a construct responsible for deploying an ECR container image
+ * for the tile server test.
```
Comment (edparris) [inline lib/osml/data_intake/di_container.ts:59]: NIT: "tile server test" cut and paste mistake.

File: lib/osml/data_intake/roles/di_lambda_role.ts (hunk 3 above)
Comment (drduhe) [reply lib/osml/data_intake/roles/di_lambda_role.ts]: fixced

File: lib/osml/data_intake/di_dataplane.ts (hunk 1 above)
Comment (drduhe) [reply lib/osml/data_intake/di_dataplane.ts:37]: Fixed

File: lib/osml/data_intake/di_dataplane.ts (hunk 2 above)
Comment (drduhe) [reply lib/osml/data_intake/di_dataplane.ts]: done

File: lib/osml/data_intake/di_container.ts [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+import { RemovalPolicy, SymlinkFollowMode } from "aws-cdk-lib";
+import { DockerImageCode } from "aws-cdk-lib/aws-lambda";
+import { Construct } from "constructs";
+
+import { OSMLAccount } from "../osml_account";
+import { OSMLECRDeployment } from "../osml_ecr_deployment";
+import { OSMLVpc } from "../osml_vpc";
+
+/**
+ * Represents the configuration for the TSTestContainer Construct.
```
Comment (RanbirAulakh) [inline lib/osml/data_intake/di_container.ts:14]: nit: TSTestContainer

File: lib/osml/data_intake/di_container.ts [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    super(scope, id);
+
+    // Set the removal policy based on the provided properties
+    this.removalPolicy = props.account.prodLike
+      ? RemovalPolicy.RETAIN
+      : RemovalPolicy.DESTROY;
+
+    // Check if a custom configuration was provided: otherwise, create a new default configuration.
+    if (props.config != undefined) {
+      // Import the existing MRContainer configuration
```
Comment (RanbirAulakh) [inline lib/osml/data_intake/di_container.ts:83]: nit: MRContainer

File: lib/osml/model_runner/roles/mr_execution_role.ts [hunk 7]
```diff
@@ -18,132 +19,102 @@ import { MRContainerConfig } from "../mr_container";
...
   constructor(scope: Construct, id: string, props: MRExecutionRoleProps) {
     super(scope, id);
 
-    // Defining constants for better readability
-    const MR_CONTAINER_REPOSITORY_NAME =
-      this.mrContainerConfig.MR_CONTAINER_REPOSITORY;
-    const MR_FIRELENS_LOG_GROUP_NAME = `/aws/${this.mrDataplaneConfig.METRICS_NAMESPACE}/MRFireLens`;
-    const MR_SERVICE_LOG_GROUP_NAME = `/aws/${this.mrDataplaneConfig.METRICS_NAMESPACE}/MRService`;
+    const containerRepositoryName = new MRContainerConfig()
+      .MR_CONTAINER_REPOSITORY;
```
Comment (RanbirAulakh) [inline lib/osml/model_runner/roles/mr_execution_role.ts:57]: Do we need this? We can remove these lines and do the `*` for resources

File: lib/osml/model_runner/roles/mr_task_role.ts [hunk 8]
```diff
@@ -39,7 +39,7 @@ export interface MRTaskRoleProps {
 }
 
 /**
- * Represents an MRTaskRole construct.
+ * Represents aa MRTaskRole construct.
```
Comment (RanbirAulakh) [inline lib/osml/model_runner/roles/mr_task_role.ts:42]: Typo

File: test/test_account.ts [hunk 9]
```diff
@@ -20,5 +21,6 @@ export const test_special_account: OSMLAccount = {
   region: "us-isob-east-1",
   isAdc: true,
   deployModelRunner: true,
-  deployTileServer: true
+  deployTileServer: true,
+  deployDataIntake: true
```
Comment (RanbirAulakh) [inline test/test_account.ts:25]: Could you update the README.md in `lib/accounts` in the guidance repo to reflect on this?

File: lib/osml/data_intake/di_container.ts (hunk 5 above)
Comment (drduhe) [reply lib/osml/data_intake/di_container.ts]: fixed

File: lib/osml/model_runner/roles/mr_execution_role.ts (hunk 7 above)
Comment (drduhe) [reply lib/osml/model_runner/roles/mr_execution_role.ts:57]: Do this in the follow up PR to clean up roles

File: lib/osml/model_runner/roles/mr_task_role.ts (hunk 8 above)
Comment (drduhe) [reply lib/osml/model_runner/roles/mr_task_role.ts:42]: fixing

File: test/test_account.ts (hunk 9 above)
Comment (drduhe) [reply test/test_account.ts:25]: Will do in the Guidance PR

**Review by RanbirAulakh — APPROVED** [review]

**Pull request merged**